File: src/main/java/com/example/teoria/financiamento/Financiamento.java
package com.example.teoria.financiamento;

public class Financiamento {
    public Double valorTotal;
    public Double entrada;
    public Integer parcelas;    

    public Financiamento(Double valorTotal, Double entrada, Integer parcelas) {
        if (entrada < valorTotal * 0.2) {
            throw new RuntimeException("A entrada deve ser pelo menos 20% do valor total");
            }
        else if (parcelas < 6) {
            throw new RuntimeException("O número de parcelas deve ser, no mínimo, 6");
        }
        this.valorTotal = valorTotal;
        this.entrada = entrada;
        this.parcelas = parcelas;
    }
    public double prestacao() {
        return (valorTotal - entrada) / parcelas;
    }

}

File: src/main/java/com/example/python/projeto2/Financiamento.py
class Financiamento:
    def __init__ (self, valorTotal, entrada, parcelas):
        if entrada < valorTotal * 0.2:
            raise ValueError("A entrada deve ser superior a 20% do valor total do financiamento")
        elif parcelas < 6:
            raise ValueError("O número de parcelas deve ser no mínimo seis")
    
        self.valorTotal = valorTotal
        self.entrada = entrada
        self.parcelas = parcelas

    def prestacao(self):
        return (self.valorTotal-self.entrada)/self.parcelas

finan = Financiamento(1000, 300, 5)
print(finan.prestacao())